pipeline/objects/graph.py

```diff
@@ -49,39 +50,91 @@ def __init__(
         self.outputs = outputs if outputs is not None else []
         self.nodes = nodes if nodes is not None else []
         self.models = models if models is not None else []
+        # Flag set when all functions with the on_startup field have run
+        self._has_run_startup = False
         self.compute_type = compute_type
         self.min_gpu_vram_mb = min_gpu_vram_mb
         # Flag set when all models have had their `load()` methods called
-        self._loaded = False
 
-    def _load(self):
-        if self._loaded:
+    def _startup(self):
+        if self._has_run_startup:
             return
-        for model in self.models:
-            # TODO check dir of model.model
-            if hasattr(model.model, "load"):
-                print("Loading model (%s)" % model.local_id)
-                model.model.load()
+
+        startup_variables = {}
+
+        for var in self.variables:
+            # At the moment only the PipelineFile variable can be used on startup
+            if isinstance(var, PipelineFile):
+                startup_variables[var.local_id] = var
+
+        for node in self.nodes:
+
+            node_inputs: List[Variable] = []
+            node_function: Function = None
+
+            for function in self.functions:
+                if function.local_id == node.function.local_id:
+                    node_function = function
+                    break
+            if (
+                hasattr(node_function.function, "__run_once__")
+                and node_function.function.__run_once__
```

Comment:
Think you can simplify this and the others, e.g.
```suggestion
                getattr(node_function.function, "__run_once__", False)
```

Reply:
Good suggestion! Noob mistake by me